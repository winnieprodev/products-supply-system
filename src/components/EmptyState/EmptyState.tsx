import { SadIcon } from "components";
import React from "react";

import * as Styled from "./emptyState.styled";

interface OwnProps {
  items: any[];
  message?: string;
  searchPhrase?: string;
  searchMessage?: string;
}
export const EmptyState: React.FC<React.PropsWithChildren<OwnProps>> = ({
  items,
  children,
  message = "No items available",
  searchPhrase,
  searchMessage = "No items found",
}) => {
  if (items.length > 0) {
    return <>{children}</>;
  }
  return (
    <Styled.Wrapper>
      <Styled.Icon>
        <SadIcon />
      </Styled.Icon>
      <Styled.Title>{searchPhrase ? searchMessage : message}</Styled.Title>
    </Styled.Wrapper>
  );
};

export default EmptyState;
