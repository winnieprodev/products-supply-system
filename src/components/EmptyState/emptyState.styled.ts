import styled, { keyframes } from "styled-components";

export const Wrapper = styled.div`
  margin: 10vh auto 0;
  width: 260px;
`;

const Appear = keyframes`
  from {
    transform: scale(1.5);
    opacity: 0;
  }
  to {
    transform: scale(1);
    opacity: 0.1;
  }
`;

export const Icon = styled.div`
  text-align: center;
  font-size: 120px;
  opacity: 0.1;
  animation: ${Appear} ease 0.5s;
  margin: 4px;
`;

export const Title = styled.div`
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 16px;
`;
