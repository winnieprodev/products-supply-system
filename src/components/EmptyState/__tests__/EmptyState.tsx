import { render } from "@testing-library/react";

import EmptyState from "../EmptyState";

describe("EmptyState", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for zero items", () => {
      const { container } = render(
        <EmptyState items={[]}>do not show that!</EmptyState>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for zero items with custom message", () => {
      const { container } = render(
        <EmptyState items={[]} message="test of a custom message">
          do not show that!
        </EmptyState>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for one item with custom message", () => {
      const { container } = render(
        <EmptyState items={["item"]}>Show me that!</EmptyState>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for zero items when the searchQuery is used", () => {
      const { container } = render(
        <EmptyState items={[]} searchPhrase="some">
          Show me that!
        </EmptyState>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for zero items with custom message, when the searchQuery is used", () => {
      const { container } = render(
        <EmptyState
          items={[]}
          searchPhrase="some"
          searchMessage="custom message"
        >
          Show me that!
        </EmptyState>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
