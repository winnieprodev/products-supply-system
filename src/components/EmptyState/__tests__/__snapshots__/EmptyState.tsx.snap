// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Snapshot tests it matches the snapshot for one item with custom message 1`] = `Show me that!`;

exports[`EmptyState Snapshot tests it matches the snapshot for zero items 1`] = `
.c0 {
  margin: 10vh auto 0;
  width: 260px;
}

.c1 {
  text-align: center;
  font-size: 120px;
  opacity: 0.1;
  -webkit-animation: ljPNfY ease 0.5s;
  animation: ljPNfY ease 0.5s;
  margin: 4px;
}

.c2 {
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sad-tear fa-w-16 "
      data-icon="sad-tear"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zM152 416c-26.5 0-48-21-48-47 0-20 28.5-60.4 41.6-77.8 3.2-4.3 9.6-4.3 12.8 0C171.5 308.6 200 349 200 369c0 26-21.5 47-48 47zm16-176c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm170.2 154.2C315.8 367.4 282.9 352 248 352c-21.2 0-21.2-32 0-32 44.4 0 86.3 19.6 114.7 53.8 13.8 16.4-11.2 36.5-24.5 20.4z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    No items available
  </div>
</div>
`;

exports[`EmptyState Snapshot tests it matches the snapshot for zero items when the searchQuery is used 1`] = `
.c0 {
  margin: 10vh auto 0;
  width: 260px;
}

.c1 {
  text-align: center;
  font-size: 120px;
  opacity: 0.1;
  -webkit-animation: ljPNfY ease 0.5s;
  animation: ljPNfY ease 0.5s;
  margin: 4px;
}

.c2 {
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sad-tear fa-w-16 "
      data-icon="sad-tear"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zM152 416c-26.5 0-48-21-48-47 0-20 28.5-60.4 41.6-77.8 3.2-4.3 9.6-4.3 12.8 0C171.5 308.6 200 349 200 369c0 26-21.5 47-48 47zm16-176c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm170.2 154.2C315.8 367.4 282.9 352 248 352c-21.2 0-21.2-32 0-32 44.4 0 86.3 19.6 114.7 53.8 13.8 16.4-11.2 36.5-24.5 20.4z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    No items found
  </div>
</div>
`;

exports[`EmptyState Snapshot tests it matches the snapshot for zero items with custom message 1`] = `
.c0 {
  margin: 10vh auto 0;
  width: 260px;
}

.c1 {
  text-align: center;
  font-size: 120px;
  opacity: 0.1;
  -webkit-animation: ljPNfY ease 0.5s;
  animation: ljPNfY ease 0.5s;
  margin: 4px;
}

.c2 {
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sad-tear fa-w-16 "
      data-icon="sad-tear"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zM152 416c-26.5 0-48-21-48-47 0-20 28.5-60.4 41.6-77.8 3.2-4.3 9.6-4.3 12.8 0C171.5 308.6 200 349 200 369c0 26-21.5 47-48 47zm16-176c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm170.2 154.2C315.8 367.4 282.9 352 248 352c-21.2 0-21.2-32 0-32 44.4 0 86.3 19.6 114.7 53.8 13.8 16.4-11.2 36.5-24.5 20.4z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    test of a custom message
  </div>
</div>
`;

exports[`EmptyState Snapshot tests it matches the snapshot for zero items with custom message, when the searchQuery is used 1`] = `
.c0 {
  margin: 10vh auto 0;
  width: 260px;
}

.c1 {
  text-align: center;
  font-size: 120px;
  opacity: 0.1;
  -webkit-animation: ljPNfY ease 0.5s;
  animation: ljPNfY ease 0.5s;
  margin: 4px;
}

.c2 {
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sad-tear fa-w-16 "
      data-icon="sad-tear"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zM152 416c-26.5 0-48-21-48-47 0-20 28.5-60.4 41.6-77.8 3.2-4.3 9.6-4.3 12.8 0C171.5 308.6 200 349 200 369c0 26-21.5 47-48 47zm16-176c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm170.2 154.2C315.8 367.4 282.9 352 248 352c-21.2 0-21.2-32 0-32 44.4 0 86.3 19.6 114.7 53.8 13.8 16.4-11.2 36.5-24.5 20.4z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    custom message
  </div>
</div>
`;
